test/meta_testing_tools.jl

```diff
@@ -2,53 +2,57 @@
 # if they were less nasty in implementation we might consider moving them to a package
 # MetaTesting.jl
 
-# need to bring this into scope explictly so can use in @testset nonpassing_results
-using Test: DefaultTestSet
-
 """
-    nonpassing_results(f)
+    EncasedTestset(desc, results) <: AbstractTestset
 
-`f` should be a function that takes no argument, and calls some code that used `@test`.
-Invoking it via `nonpassing_results(f)` will prevent those `@test` being added to the
-current testset, and will return a collection of all nonpassing test results.
+A custom testset that encases all test results within, not letting them out.
+It doesn't let anything propagate up to the parent testset
+(or to the top-level fallback testset, which throws an error on any non-passing result).
+Not passes, not failures, not even errors.
+
+
+This is useful for being able to observe the testsets results programatically;
+without them triggering actual passes/failures/errors.
 """
-function nonpassing_results(f)
-    mute() do
-        nonpasses = []
-        # Specify testset type incase parent testset is some other typer
-        @testset DefaultTestSet "nonpassing internal" begin
-            f()
-            ts = Test.get_testset()  # this is the current testset "nonpassing internal"
-            nonpasses = _extract_nonpasses(ts)
-            # Prevent the failure being recorded in parent testset.
-            empty!(ts.results)
-            ts.anynonpass = false
-        end
-        # Note: we allow the "nonpassing internal" testset to still be pushed as an empty
-        # passing testset in its parent testset. We could remove that if we wanted
-        return nonpasses
+struct EncasedTestset <: Test.AbstractTestSet
```

Comment:
Naming convention seems to be `TestSet`
```suggestion
struct EncasedTestSet <: Test.AbstractTestSet
```
(maybe worth being conistent, in case we move this to its own package in future)

anyway, nitpick aside, i think this is good to go!